decidim-core/app/resolvers/decidim/meta_image_url_resolver.rb

```diff
@@ -0,0 +1,109 @@
+# frozen_string_literal: true
+
+module Decidim
+  # Resolves the most appropriate image URL for meta tags from a given resource within an organization.
+  # Searches in the following order: attachment images, images in resource description, participatory space images, and default content block images.
+  # Ensures the selected image is resized to 1200x630 pixels.
+  class MetaImageUrlResolver
+    include TranslatableAttributes
+
+    def initialize(resource, current_organization)
+      @resource = resource
+      @current_organization = current_organization
+    end
+
+    # Resolves the image URL to be used for meta tags.
+    #
+    # @return [String, nil] - The resolved image URL or nil if no image is found.
+    def resolve
+      blob = determine_image_blob.presence
+      resize_image_blob(blob) if blob.present?
+    end
+
+    private
+
+    # Determines the image blob to be used for meta tags by following a hierarchy.
+    #
+    # @return [ActiveStorage::Blob, nil] - The image blob or nil if no image is found.
+    def determine_image_blob
+      blob_from_attachment_image || blob_from_description(@resource) || blob_from_participatory_space || blob_from_content_blocks
```

Comment:
I'm thinking that we should try something more generic.
In this example an attachment done over an assembly would use that as image, and it should really be the hero image first. no?

```suggestion
      blob_from_image_fields || blob_from_attachment_image || blob_from_description(@resource) || blob_from_participatory_space || blob_from_content_blocks
```

in here the `blob_from_image_fields` should check for the same think done in `blob_from_participatory_space`
```ruby
 blob = participatory_space.try(:hero_image).try(:blob) || participatory_space.try(:banner_image).try(:blob)
```
Maybe check if it can be more cases, for instance, if visiting the profile page for an user, should we show the `avatar` (that would be a "image field")